packages/clay-date-picker/src/ClayDatePicker.js

```diff
@@ -0,0 +1,384 @@
+import PropTypes from 'prop-types';
+import React, {useState, useEffect, useRef} from 'react';
+
+import {useCurrentMonth, useCurrentTime, useWeeks} from './Hooks';
+
+import Button from './Button';
+import DateNavigation from './DateNavigation';
+import Day from './Day';
+import Dropdown from './Dropdown';
+import InputDate from './InputDate';
+import TimePicker from './TimePicker';
+import Week from './Week';
+import Weekday from './Weekday';
+import Weekdays from './Weekdays';
+
+/**
+ * ClayDatePicker component.
+ * @return {React.createElement}
+ */
+function ClayDatePicker({
+	ariaLabel,
+	dateFormat,
+	defaultValue,
+	expanded: initialExpanded,
+	firstDayOfWeek,
+	footerElement,
+	id,
+	initialMonth,
+	inputName,
+	months,
+	onValueChange,
+	placeholder,
+	spritemap,
+	time,
+	timeFormat,
+	useNative,
+	weekdaysShort,
+	years,
+}) {
+	/**
+	 * Indicates the current month rendered on the screen.
+	 * @type {!Date}
+	 */
+	const [currentMonth, setCurrentMonth] = useCurrentMonth(initialMonth);
+
+	/**
+	 * Indicates the time selected by the user.
+	 * @type {!Date}
+	 */
+	const [currentTime, setCurrentTime] = useCurrentTime(timeFormat);
+
+	/**
+	 * The day selected by the user.
+	 * @type {!Date}
+	 */
+	const [daySelected, setDaySelected] = useState(initialMonth);
+
+	/**
+	 * The input value.
+	 * @type {?(Date|undefined)}
+	 */
+	const [inputValue, setInputValue] = useState(undefined);
+
+	/**
+	 * An array of the weeks and days list for the current month
+	 * @type {!Array<Array<{date: Date, ?outside: bool}>>}
+	 */
+	const [weeks, setWeeks] = useWeeks(currentMonth, firstDayOfWeek);
+
+	/**
+	 * Flag to indicate if date is expanded.
+	 * @type {?bool}
+	 */
+	const [expanded, setExpanded] = useState(initialExpanded);
+
+	/**
+	 * Create a ref to store the datepicker DOM element
+	 */
+	const elementRef = useRef(null);
+
+	/**
+	 * Handles the click on element of the day
+	 * @param {!Date} date
+	 */
+	function handleDayClicked(date) {
+		setDaySelected(date);
+		setInputValue(date);
+	}
+
+	/**
+	 * Handles the input change.
+	 * @param {!Event} event
+	 */
+	function inputChange(event) {
+		const {value} = event.target;
+		const format = time ? `${dateFormat} ${time}` : dateFormat;
+		const date = moment(value, format);
+
+		if (date.isValid()) {
+			setCurrentMonth(date.toDate());
+			setDaySelected(date.toDate());
+
+			if (time) {
+				setCurrentTime(date.hours(), date.minutes());
+			}
+		}
+
+		if (onValueChange) {
+			onValueChange(value);
+		}
+
+		setInputValue(value);
+	}
+
+	/**
+	 * Handles datepicker view
+	 */
+	function handleCalendarButtonClicked() {
+		setExpanded(!expanded);
+	}
+
+	useEffect(
+		() => {
+			// It is not necessary to update the weeks when the native
+			// date is enabled. Unnecessary rendering and processing.
+			if (!useNative) {
+				setWeeks(currentMonth, firstDayOfWeek);
+			}
+		},
+		[currentMonth]
+	);
+
+	const classNames = classnames('input-group', {
+		[id]: id,
```

Comment:
Shouldn't this id be put on the element and not a className?

Reply:
Oh, that went unnoticed was not supposed to be there. Thanks!